bin/delete_pr_comment

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+# This script is used to delete comments on a pull request, and must be run in a buldkite PR step.
+#
+# Example:
+#   delete_pr_comment --github-token <token> --containing "part of a comment"
+
+# Check dependencies and print their versions for diagnosis purposes
+jq --version
+curl --version
+
+# Parse command line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --containing) COMMENT_CONTAINING="$2"; shift ;;
+        --github-token) GITHUB_TOKEN="$2"; shift ;;
+        *) echo "Unknown option: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+function require_option() {
+    if [[ -z "$2" ]]; then
+        echo "Error: $1 is required"
+        exit 1
+    fi
+}
+
+require_option '--containing' "$COMMENT_CONTAINING"
```

Comment:
Because of the `u` option in the shebang, this call will fail before `require_option` gets a chance to run if `$COMMENT_CONTAINING` is empty:

```
./bin/delete_pr_comment: line 29: COMMENT_CONTAINING: unbound variable
```

The result is the same (the script fails) but the UX is not as clear.

Reply:
Yeah I agree. Considering we have so many bash script, I thought I'll add this simply argument parsing function to give the scripts a better CLI, with `cli --foo value --bar true` instead of `cli value true`. It does its job, but the error message may not be intuitive.